### PR TITLE
[WIP]Add more bit_casts in loaders

### PR DESCRIPTION
**Pull request by ShFil119**
Nothing big.

### CODE REVIEW

File: rwengine/src/loaders/LoaderGXT.cpp [hunk 1]
```diff
@@ -8,20 +8,21 @@
...
-    std::uint32_t blocksize = *(std::uint32_t *)data;
+    std::uint32_t blocksize = bit_cast<std::uint32_t>(*data);
 
     data += 4;
 
     auto tdata = data + blocksize + 8;
 
     for (size_t t = 0; t < blocksize / 12; ++t) {
-        size_t offset = *(std::uint32_t *)(data + (t * 12 + 0));
+        size_t offset = bit_cast<std::uint32_t>(data + (t * 12 + 0));
```
Comment (JayFoxRox) [inline rwengine/src/loaders/LoaderGXT.cpp]: Aren't semantics changed here? I'm not familiar with the `bit_cast` function, but above you dereference data and pass that to the `bit_cast`. Here, you just calculate the address, don't dereference it and pass it to `bit_cast`

File: rwengine/src/loaders/LoaderCOL.cpp [hunk 2]
```diff
@@ -48,19 +49,19 @@ bool LoaderCOL::load(const std::string& path) {
         model->modelid = head.modelid;
 
         auto readFloat = [&]() {
-            auto f = (float*)d;
+            auto f = bit_cast<float>(*d);
```
Comment (JayFoxRox) [inline rwengine/src/loaders/LoaderCOL.cpp]: nit: `d` is `auto`, so if it's suddenly not `sizeof(float)` it *looks* like you've changed semantics. This is actually poor design in the `bit_cast` function in my opinion.

File: rwengine/src/loaders/LoaderGXT.cpp (hunk 1 above)
Comment (ShFil119) [reply rwengine/src/loaders/LoaderGXT.cpp]: You're right, I'll rebase.

(Casting ptr and dereferencing can cause UB, if padding is different. bit_cast handles it)

Comment (JayFoxRox) [inline rwengine/src/loaders/LoaderGXT.cpp]: nit: who thought it was a good idea to do (presumably) pointer arithmetic on an `auto` variable?

File: rwengine/src/loaders/LoaderCOL.cpp (hunk 2 above)
Comment (ShFil119) [reply rwengine/src/loaders/LoaderCOL.cpp]: I'll add better docs.

File: rwengine/src/loaders/LoaderGXT.cpp (hunk 1 above)
Comment (paulsapps) [reply rwengine/src/loaders/LoaderGXT.cpp]: This whole thing seems a bit smelly IMO.. would it not be better to have a BinaryStream object or something that you can Seek/Read(T) etc instead?

Comment (ShFil119) [reply rwengine/src/loaders/LoaderGXT.cpp]: Do you think about treating it as stream?

**Comment by ShFil119** [comment]
I'll close it for now, I'll try later convert BinaryStream to safer ifstream.

**Pull request closed without merging**